File: lib/ui/achievements/xp_level_page.dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:go_router/go_router.dart';
import '../../core/app_constants.dart';
import '../../providers/stats_providers.dart';

/// 等级详情页 - V1.0 重构
///
/// 大圆圈显示当前等级数字 + 等级名称 + XP 进度条 (当前/下一级)。
/// 下方列出所有等级, 当前等级高亮。
class XpLevelPage extends ConsumerWidget {
  const XpLevelPage({super.key});

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final stats = ref.watch(userStatsProvider).valueOrNull;
    final currentLevel = ref.watch(currentLevelProvider);
    final nextLevel = ref.watch(nextLevelProvider);
    final progress = ref.watch(levelProgressProvider);
    final theme = Theme.of(context);

    final totalXp = stats?.totalXp ?? 0;

    return PopScope(
      canPop: false,
      onPopInvokedWithResult: (didPop, _) {
        if (!didPop) context.go('/achievements');
      },
      child: Scaffold(
        appBar: AppBar(
          title: const Text('等级'),
          leading: IconButton(
            icon: const Icon(Icons.arrow_back),
            onPressed: () => context.go('/achievements'),
          ),
        ),
        body: ListView(
        padding: const EdgeInsets.all(16),
        children: [
          // === 当前等级大圆 ===
          Card(
            child: Padding(
              padding:
                  const EdgeInsets.symmetric(vertical: 24, horizontal: 16),
              child: Column(
                children: [
                  Container(
                    width: 96,
                    height: 96,
                    decoration: BoxDecoration(
                      shape: BoxShape.circle,
                      gradient: LinearGradient(
                        colors: [
                          theme.colorScheme.primary,
                          theme.colorScheme.tertiary,
                        ],
                      ),
                      boxShadow: [
                        BoxShadow(
                          color:
                              theme.colorScheme.primary.withValues(alpha: 0.3),
                          blurRadius: 16,
                          offset: const Offset(0, 4),
                        ),
                      ],
                    ),
                    child: Center(
                      child: Text(
                        '${currentLevel.level}',
                        style: const TextStyle(
                          color: Colors.white,
                          fontWeight: FontWeight.bold,
                          fontSize: 36,
                        ),
                      ),
                    ),
                  ),
                  const SizedBox(height: 14),
                  Text(
                    currentLevel.title,
                    style: theme.textTheme.headlineSmall
                        ?.copyWith(fontWeight: FontWeight.bold),
                  ),
                  const SizedBox(height: 6),
                  Text(
                    '$totalXp XP',
                    style: theme.textTheme.titleMedium?.copyWith(
                      color: theme.colorScheme.primary,
                      fontWeight: FontWeight.bold,
                    ),
                  ),
                  const SizedBox(height: 16),

                  // XP 进度条: 当前 XP / 下一级 XP
                  if (nextLevel != null) ...[
                    Row(
                      mainAxisAlignment: MainAxisAlignment.spaceBetween,
                      children: [
                        Text('$totalXp XP', style: theme.textTheme.labelSmall),
                        Text('${nextLevel.xpRequired} XP',
                            style: theme.textTheme.labelSmall),
                      ],
                    ),
                    const SizedBox(height: 6),
                    ClipRRect(
                      borderRadius: BorderRadius.circular(8),
                      child: LinearProgressIndicator(
                        value: progress.clamp(0.0, 1.0),
                        minHeight: 12,
                        backgroundColor:
                            theme.colorScheme.surfaceContainerHighest,
                      ),
                    ),
                    const SizedBox(height: 6),
                    Text(
                      '距 ${nextLevel.title} 还需 ${nextLevel.xpRequired - totalXp} XP',
                      style: theme.textTheme.labelSmall
                          ?.copyWith(color: theme.colorScheme.onSurfaceVariant),
                      textAlign: TextAlign.center,
                    ),
                  ] else
                    Container(
                      padding: const EdgeInsets.symmetric(
                          horizontal: 16, vertical: 8),
                      decoration: BoxDecoration(
                        color: Colors.amber.withValues(alpha: 0.15),
                        borderRadius: BorderRadius.circular(20),
                      ),
                      child: Text(
                        '已达最高等级!',
                        style: theme.textTheme.labelLarge?.copyWith(
                          color: Colors.amber.shade800,
                          fontWeight: FontWeight.bold,
                        ),
                      ),
                    ),
                ],
              ),
            ),
          ),
          const SizedBox(height: 20),

          // === 所有等级列表 ===
          Text('等级路线',
              style: theme.textTheme.titleMedium
                  ?.copyWith(fontWeight: FontWeight.bold)),
          const SizedBox(height: 8),
          ...AppLevels.levels.map((lv) {
            final isCurrent = currentLevel.level == lv.level;
            final reached = totalXp >= lv.xpRequired;
            return Card(
              color: isCurrent
                  ? theme.colorScheme.primaryContainer.withValues(alpha: 0.4)
                  : null,
              margin: const EdgeInsets.only(bottom: 6),
              child: ListTile(
                leading: Icon(
                  reached ? Icons.emoji_events : Icons.lock_outline,
                  color: reached ? Colors.amber : theme.disabledColor,
                ),
                title: Text(
                  'Lv.${lv.level} ${lv.title}',
                  style: TextStyle(
                    fontWeight:
                        isCurrent ? FontWeight.bold : FontWeight.normal,
                    color: reached ? null : theme.disabledColor,
                  ),
                ),
                subtitle: Text('${lv.xpRequired} XP'),
                trailing: isCurrent
                    ? Chip(
                        label: const Text('当前'),
                        backgroundColor: theme.colorScheme.primaryContainer,
                        labelStyle:
                            const TextStyle(fontWeight: FontWeight.bold),
                      )
                    : reached
                        ? const Icon(Icons.check_circle,
                            color: Colors.green, size: 20)
                        : null,
              ),
            );
          }),
        ],
      ),
      ),
    );
  }
}

File: app/src/main/java/com/salesquest/sales_quest/ui/achievements/XpLevelPage.kt
package com.salesquest.sales_quest.ui.achievements

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material.icons.filled.EmojiEvents
import androidx.compose.material.icons.filled.Lock
import androidx.compose.material3.AssistChip
import androidx.compose.material3.Card
import androidx.compose.material3.CenterAlignedTopAppBar
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.LinearProgressIndicator
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.material3.TopAppBarDefaults
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.lifecycle.viewmodel.compose.viewModel
import com.salesquest.sales_quest.core.AppLevels
import com.salesquest.sales_quest.core.LevelDef

/** 等级详情页 - 当前等级大圆 + XP 进度条 + 全部等级路线 */
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun XpLevelPage(
    onBack: () -> Unit = {},
    viewModel: XpLevelViewModel = viewModel()
) {
    val totalXp by viewModel.totalXp.collectAsState()

    val currentLevel = AppLevels.getLevel(totalXp)
    val nextLevel = AppLevels.getNextLevel(totalXp)
    val progress = AppLevels.getProgress(totalXp)

    Scaffold(
        topBar = {
            CenterAlignedTopAppBar(
                title = { Text("等级") },
                navigationIcon = {
                    IconButton(onClick = onBack) {
                        Icon(Icons.AutoMirrored.Filled.ArrowBack, contentDescription = "返回")
                    }
                },
                colors = TopAppBarDefaults.centerAlignedTopAppBarColors()
            )
        }
    ) { innerPadding ->
        LazyColumn(
            modifier = Modifier
                .fillMaxSize()
                .padding(innerPadding),
            contentPadding = androidx.compose.foundation.layout.PaddingValues(16.dp)
        ) {
            item {
                CurrentLevelCard(
                    level = currentLevel,
                    nextLevel = nextLevel,
                    totalXp = totalXp,
                    progress = progress
                )
            }
            item {
                Spacer(Modifier.height(20.dp))
                Text(
                    "等级路线",
                    style = MaterialTheme.typography.titleMedium,
                    fontWeight = FontWeight.Bold
                )
                Spacer(Modifier.height(8.dp))
            }
            items(AppLevels.levels, key = { it.level }) { lv ->
                LevelRow(
                    lv = lv,
                    currentLevel = currentLevel,
                    totalXp = totalXp
                )
            }
        }
    }
}

/** 当前等级卡片 */
@Composable
fun CurrentLevelCard(
    level: LevelDef,
    nextLevel: LevelDef?,
    totalXp: Int,
    progress: Double
) {
    val theme = MaterialTheme.colorScheme
    Card(modifier = Modifier.fillMaxWidth()) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(vertical = 24.dp, horizontal = 16.dp),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Box(
                modifier = Modifier
                    .size(96.dp)
                    .clip(CircleShape)
                    .background(Brush.linearGradient(listOf(theme.primary, theme.tertiary))),
                contentAlignment = Alignment.Center
            ) {
                Text(
                    "${level.level}",
                    color = Color.White,
                    fontWeight = FontWeight.Bold,
                    fontSize = androidx.compose.ui.unit.TextUnit.Unspecified,
                    style = MaterialTheme.typography.displaySmall
                )
            }
            Spacer(Modifier.height(14.dp))
            Text(
                level.title,
                style = MaterialTheme.typography.headlineSmall,
                fontWeight = FontWeight.Bold
            )
            Spacer(Modifier.height(6.dp))
            Text(
                "$totalXp XP",
                style = MaterialTheme.typography.titleMedium,
                color = theme.primary,
                fontWeight = FontWeight.Bold
            )
            Spacer(Modifier.height(16.dp))

            if (nextLevel != null) {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween
                ) {
                    Text("$totalXp XP", style = MaterialTheme.typography.labelSmall)
                    Text("${nextLevel.xpRequired} XP", style = MaterialTheme.typography.labelSmall)
                }
                Spacer(Modifier.height(6.dp))
                LinearProgressIndicator(
                    progress = { progress.coerceIn(0.0, 1.0).toFloat() },
                    modifier = Modifier
                        .fillMaxWidth()
                        .height(12.dp)
                        .clip(RoundedCornerShape(8.dp)),
                    trackColor = theme.surfaceContainerHighest
                )
                Spacer(Modifier.height(6.dp))
                Text(
                    "距 ${nextLevel.title} 还需 ${nextLevel.xpRequired - totalXp} XP",
                    style = MaterialTheme.typography.labelSmall,
                    color = theme.onSurfaceVariant
                )
            } else {
                Box(
                    modifier = Modifier
                        .clip(RoundedCornerShape(20.dp))
                        .background(Color(0xFFFFC107).copy(alpha = 0.15f))
                        .padding(horizontal = 16.dp, vertical = 8.dp)
                ) {
                    Text(
                        "已达最高等级!",
                        style = MaterialTheme.typography.labelLarge,
                        color = Color(0xFFE65100),
                        fontWeight = FontWeight.Bold
                    )
                }
            }
        }
    }
}

/** 单个等级条目 */
@Composable
fun LevelRow(lv: LevelDef, currentLevel: LevelDef, totalXp: Int) {
    val theme = MaterialTheme.colorScheme
    val isCurrent = currentLevel.level == lv.level
    val reached = totalXp >= lv.xpRequired

    Card(
        modifier = Modifier
            .fillMaxWidth()
            .padding(bottom = 6.dp),
        colors = if (isCurrent) {
            androidx.compose.material3.CardDefaults.cardColors(
                containerColor = theme.primaryContainer.copy(alpha = 0.4f)
            )
        } else {
            androidx.compose.material3.CardDefaults.cardColors()
        }
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 16.dp, vertical = 12.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            Icon(
                if (reached) Icons.Filled.EmojiEvents else Icons.Filled.Lock,
                contentDescription = null,
                tint = if (reached) Color(0xFFFFC107) else theme.outline,
                modifier = Modifier.size(24.dp)
            )
            Spacer(Modifier.width(12.dp))
            Column(modifier = Modifier.weight(1f)) {
                Text(
                    "Lv.${lv.level} ${lv.title}",
                    style = MaterialTheme.typography.bodyLarge,
                    fontWeight = if (isCurrent) FontWeight.Bold else FontWeight.Normal,
                    color = if (reached) Color.Unspecified else theme.outline
                )
                Text(
                    "${lv.xpRequired} XP",
                    style = MaterialTheme.typography.labelSmall,
                    color = theme.onSurfaceVariant
                )
            }
            when {
                isCurrent -> {
                    AssistChip(
                        onClick = {},
                        label = { Text("当前") }
                    )
                }
                reached -> {
                    Icon(
                        Icons.Filled.CheckCircle,
                        contentDescription = null,
                        tint = Color(0xFF4CAF50),
                        modifier = Modifier.size(20.dp)
                    )
                }
            }
        }
    }
}
